File: GoPostule_project.Data/Entitée/Offre.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace GoPostule_project.Data.Entitée
{
    public class Offre
    {
        public int Id { get; set; }
        public string Titre { get; set; } = string.Empty;
        public string Entreprise { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string Missions { get; set; } = string.Empty;
        public string ProfilRecherche { get; set; } = string.Empty;
        public string Competences { get; set; } = string.Empty;
        public string Localisation { get; set; } = string.Empty;
        public string Salaire { get; set; } = string.Empty;
        public string TypeContrat { get; set; } = string.Empty;
        public string Secteur { get; set; } = string.Empty;
        public DateTime? DateLimite { get; set; } //? Cela signifie que la date est optionnelle.
        public int PlacesDisponibles { get; set; } = 1;
        public bool Actif { get; set; } = true;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // Navigation
        public ICollection<Candidature> Candidature { get; set; }
            = new List<Candidature>();
    }
}

File: lib/shared/offre.dart
// lib/shared/offre.dart
class Offre {
  final int id;
  final String titre;
  final String entreprise;
  final String description;
  final String? missions;
  final String? profilRecherche;
  final String? competences;
  final String localisation;
  final String? salaire;
  final String typeContrat;
  final String secteur;
  final String? dateLimite;
  final int? placesDisponibles;
  final bool actif;
  final DateTime? createdAt;

  Offre({
    required this.id,
    required this.titre,
    required this.entreprise,
    required this.description,
    this.missions,
    this.profilRecherche,
    this.competences,
    required this.localisation,
    this.salaire,
    required this.typeContrat,
    required this.secteur,
    this.dateLimite,
    this.placesDisponibles,
    required this.actif,
    this.createdAt,
  });

  // ✅ CONSTRUCTEUR JSON → OBJET (clés backend en MAJUSCULE)
  factory Offre.fromJson(Map<String, dynamic> json) {
    return Offre(
      id: json['Id'] ?? 0,
      titre: json['Titre'] ?? '',
      entreprise: json['Entreprise'] ?? '',
      description: json['Description'] ?? '',
      missions: json['Missions'],
      profilRecherche: json['ProfilRecherche'],
      competences: json['Competences'],
      localisation: json['Localisation'] ?? '',
      salaire: json['Salaire'],
      typeContrat: json['TypeContrat'] ?? '',
      secteur: json['Secteur'] ?? '',
      dateLimite: json['DateLimite'],
      placesDisponibles: json['PlacesDisponibles'],
      actif: json['Actif'] ?? true,
      createdAt: json['CreatedAt'] != null
          ? DateTime.tryParse(json['CreatedAt'])
          : null,
    );
  }

  // ✅ OBJET → JSON (pour les envois vers l'API)
  Map<String, dynamic> toJson() => {
    'Titre': titre,
    'Entreprise': entreprise,
    'Description': description,
    'Localisation': localisation,
    'TypeContrat': typeContrat,
    'Secteur': secteur,
    'Missions': missions,
    'Competences': competences,
    'Salaire': salaire,
    'DateLimite': dateLimite,
    'PlacesDisponibles': placesDisponibles,
    'Actif': actif,
  };
}